arithmetization/src/main/java/net/consensys/linea/plugins/rpc/tracegeneration/GenerateConflatedTracesV2.java

```diff
@@ -109,32 +113,41 @@ private TraceService getTraceService() {
                     "Unable to find trace service. Please ensure TraceService is registered."));
   }
 
+  @SneakyThrows(IOException.class)
   private String writeTraceToFile(
       final ZkTracer tracer, final TraceRequestParams traceRequestParams) {
-    final Path fileName = generateOutputFileName(traceRequestParams);
-    tracer.writeToFile(fileName);
-    return fileName.toAbsolutePath().toString();
+    // Generate the original and final trace file name.
+    final String origTraceFileName = generateOutputFileName(traceRequestParams);
+    // Generate and resolve the original and final trace file path.
+    final Path origTraceFilePath = generateOutputFilePath(origTraceFileName + TRACE_FILE_EXTENSION);
+
+    // Write the trace at the original and final trace file path, but with the suffix .tmp at the
+    // end of the file.
+    final Path tmpTraceFilePath =
+        tracer.writeToTmpFile(tracesOutputPath, origTraceFileName + ".", TRACE_TEMP_FILE_EXTENSION);
```

Comment:
don't we have a '.' too many?

I would have found it easier to follow with final_name = xxx-yyy.lt

and

temp_name = final_name + ".tmp"

Reply:
No, this dot is needed because `Files.createTempFile(rootDir, prefix, suffix)` appends <timestamp>.<suffix> at the end and there is no dot symbol before the timestamp so with this approach I am setting the suffix to be `.lt.tmp` and the renamed file will end with only `.lt`. Here is a real example from my local testing:

```
// Temp file name
600-700.conflated.0.6.0-rc.16783354941660875494.lt.tmp

// Final file name
600-700.conflated.0.6.0-rc1.lt
```